### ISSUE TITLE
ArrayHash: malformed array with integer keys

### ISSUE THREAD

**Issue by milo** [nette/utils#65]
When using ArrayHash created from array with integer keys, following conversion to `(array)` creates malformed array with string keys. Such array cannot be tested by `isset()` and `array_key_exists()`, it will results to `FALSE`.

This is potentially dangerous with FormContainer and integer components:
```
$cont = $form->addContaner('foo');
$cont->addText(1, 'One');
$cont->addText(12, 'One');
```

Opened PHP BUG [#69799](https://bugs.php.net/bug.php?id=69799), details there.

**Comment by dg** [comment]
It is known PHP limitation https://bugs.php.net/bug.php?id=51682

**Comment by milo** [comment]
@dg Hm, they already closed my bug. And now I'm reading your. Thanks for the link.

OT:
And the reason: "Fixing that implies a perfomance decrease, hence the better seems be keep it as an known issue, but documented.". So, we can create invalid arrays in PHP, without single warning or notice. Primarily, the PHP7 is the fastest as never the PHP was.

Just, one, word:
IT IS A SHIT!
==

Wait, it is four words. Yeah, who cares.

**Comment by JanTvrdik** [comment]
Its because integer properties are hashed as strings in object and as integers in arrays. AFAIK there is no way around it without degrading performance.

**Comment by dg** [comment]
The same problem as here https://github.com/nette/forms/pull/77

**Comment by enumag** [comment]
If `(array)` is not safe to use and there is a workaround, maybe we should add a `ArrayHash::toArray()` method.

**Comment by fprochazka** [comment]
@enumag :+1:

**Comment by milo** [comment]
IIRC, there was a decision to extends `stdClass` instead of `ArrayObject` due to deep `isset()` bug. If so, maybe there is [a chance](https://bugs.php.net/bug.php?id=62059#1431523879) sice PHP 5.5.16.

The bug is not solved completely, but it works if you don't overload the `offsetGet()` method.

**Comment by dg** [comment]
Workaround for forms exists: getValues(TRUE)

**Comment by enumag** [comment]
@dg Yes, I'm aware.

**Comment by fprochazka** [comment]
@dg just curiouse, are you against toArray() method? I think it's usefull

**Comment by dg** [comment]
I am not against.

**Comment by milo** [comment]
Didn't find the corresponding PHP commit, but according to https://www.slideshare.net/nikita_ppv/php-language-trivia, slide 27, this will be fixed in PHP 7.2.
